worker/src/enclave/worker_api_direct_server.rs

```diff
@@ -210,66 +209,101 @@ pub fn handle_direct_invocation_request(req: DirectWsServerRequest) -> Result<()
+        let status_update = TrustedOperationStatus::decode(&mut status_update_slice).unwrap();
+        let mut hash_slice = slice::from_raw_parts(hash_encoded, hash_size as usize);
+        if let Ok(hash) = Hash::decode(&mut hash_slice) {
+            // Aquire watched list lock
+            let mutex = load_watched_list().unwrap();
+            let mut watch_list = mutex.lock().unwrap();
+            let mut continue_watching = true;
+            if let Some(client_event) = watch_list.get_mut(&hash) {
+                let mut event = &mut client_event.response;
+                // Aquire result of old RpcResponse
+                let old_result: Vec<u8> = event.result.clone();
+                let mut result = RpcReturnValue::decode(&mut old_result.as_slice()).unwrap();
+
+                match status_update {
+                    TrustedOperationStatus::Invalid
+                    | TrustedOperationStatus::InSidechainBlock(_)
+                    | TrustedOperationStatus::Finalized
+                    | TrustedOperationStatus::Usurped => {
+                        // Stop watching
+                        result.do_watch = false;
+                        continue_watching = false;
+                    }
+                    _ => {}
+                };
+                // update response
+                result.status = DirectRequestStatus::TrustedOperationStatus(status_update);
+                event.result = result.encode();
+                client_event
+                    .client
+                    .send(serde_json::to_string(&event).unwrap())
+                    .unwrap();
+
+                if !continue_watching {
+                    client_event.client.close(CloseCode::Normal).unwrap();
                 }
-                _ => {}
-            };
-            // update response
-            result.status = DirectRequestStatus::TrustedOperationStatus(status_update);
-            event.result = result.encode();
-            client_event
+            } else {
+                continue_watching = false;
+            }
+            if !continue_watching {
+                watch_list.remove(&hash);
+            }
+        }
+    */
+    sgx_status_t::SGX_SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ocall_send_response_with_uuid(
+    request_id_encoded: *const u8,
+    request_id_size: u32,
+    uuid_encoded: *const u8,
+    uuid_size: u32,
+) -> sgx_status_t {
+    let mut request_id_slice = slice::from_raw_parts(request_id_encoded, request_id_size as usize);
+    let uuid_slice = slice::from_raw_parts(uuid_encoded, uuid_size as usize);
+    if let Ok(request_id) = u128::decode(&mut request_id_slice) {
+        let mutex = if let Some(mutex) = load_watched_list() {
+            mutex
+        } else {
+            return sgx_status_t::SGX_ERROR_UNEXPECTED;
+        };
+        let mut guard = mutex.lock().unwrap();
+        if let Some(client_response) = guard.get_mut(&request_id) {
+            // TODO @Bigna do we need this?
```

Comment:
@brenzi  please check

Reply:
Hm, it depends I guess.

The currently implemented client is stateless and always expects the same answer type. So it does not know if it's an answer containing only the RequestId or a response containing the OrderUUID, or an answer from a Cancel Order, or whatever request, it always decodes it the same way (and as it's originally an json rpc request, it also expects an json rpc response, which looks like this:
` RpcResponse { jsonrpc: "2.0", result: [181, 21, 173, 21, ..., 125, 0, 0], id: 1 }` )

As our client is for debugging purposes only, I think we should keep it stateless for now - but that would mean, you need to return an `RpcResponse`:

```rust
RpcResponse {
      result: result.encode(),
      jsonrpc: full_rpc_response.jsonrpc.clone(), // in our client this is not checked
      id: full_rpc_response.id, // in json rpc this id should be the same as the request id. But we're currently not using it
}
```
where the result is a RpcReturnValue struct:
`let result = RpcReturnValue::new(orderuuid_slice.to_vec(), false, DirectRequestStatus::Ok);`

So - our client does currently not check the response id nor the jsonrpc version. So you have two Options:
1) Enter dummy values
2) Do it the same way I did in function `ocall_send_status`, namely, storing the first jsonrpc response from the enclave (see line 128), and upon ocall load the original rpc response and modify only the result.

Reply:
I guess it's up to you to decide how you want to receive an reponse, as the main purpose is benchmarking

Reply:
In this issue you find some logs, which might help to understand what exactly is happening here: https://github.com/integritee-network/worker/issues/202

Especially interesting it's getting after the `[Enclave] Event!` Log (Event => `ocall_update_status_event`)

Reply:
I implemented it @haerdib ... could you please verify it again